README.md
# nos1000jours-web-pro

## Modifier le site

Cloner le repo puis :

```sh
yarn
yarn dev
```

## Pour exécuter les tests :

```
yarn test
```


CHANGELOG.md
## [1.12.1](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.12.0...v1.12.1) (2023-05-30)


### Bug Fixes

* mention legal ([a4563f4](https://github.com/SocialGouv/nos1000jours-web-pro/commit/a4563f4b642482ea82e62b150ba14ee78b617675))

# [1.12.0](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.11.3...v1.12.0) (2023-05-23)


### Bug Fixes

* port ([1f5689b](https://github.com/SocialGouv/nos1000jours-web-pro/commit/1f5689b8aa32944414e98e0df082b0482bfd03fe))
* re-add release workflow ([ac6f0eb](https://github.com/SocialGouv/nos1000jours-web-pro/commit/ac6f0eb7fe18791e31c3f6bca41c66d0b70b0827))


### Features

* ci on kontinuous ([b0ca233](https://github.com/SocialGouv/nos1000jours-web-pro/commit/b0ca23375009e302a85635b5d794c6380ae66531))
* ci on kontinuous ([31045de](https://github.com/SocialGouv/nos1000jours-web-pro/commit/31045dec3bd3c9f063e402e980d747fe74dfe48f))

## [1.11.3](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.11.2...v1.11.3) (2022-04-12)


### Bug Fixes

* use kube-workflow ([#73](https://github.com/SocialGouv/nos1000jours-web-pro/issues/73)) ([92b8324](https://github.com/SocialGouv/nos1000jours-web-pro/commit/92b83241c5504877b2ab6643a9a0537ccca23f2e))

## [1.11.2](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.11.1...v1.11.2) (2022-03-14)


### Bug Fixes

* lints ([#70](https://github.com/SocialGouv/nos1000jours-web-pro/issues/70)) ([6f4cca8](https://github.com/SocialGouv/nos1000jours-web-pro/commit/6f4cca841e7773eee890e9e09da44051b766511e))

## [1.11.1](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.11.0...v1.11.1) (2022-02-24)


### Bug Fixes

* **ci:** autodevops-helm ([9258d15](https://github.com/SocialGouv/nos1000jours-web-pro/commit/9258d15896a387bb1f76ca3f8e1ea370ca48b693))
* **ci:** autodevops-helm ([8234418](https://github.com/SocialGouv/nos1000jours-web-pro/commit/8234418583ff60e8f7d8238834b5a8879fbb439f))
* **ci:** autodevops-helm ([0ba29e2](https://github.com/SocialGouv/nos1000jours-web-pro/commit/0ba29e2c8a29245f2b81deeb4346c931e9d3abdf))

# [1.11.0](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.10.5...v1.11.0) (2022-02-10)


### Features

* **traduction:** adaptation questions + reponses ([c6bd6e4](https://github.com/SocialGouv/nos1000jours-web-pro/commit/c6bd6e45996d84b13320a51949556dd459de4c44))
* **traduction:** ajout de la consigne pour les langues étrangères ([aa0b577](https://github.com/SocialGouv/nos1000jours-web-pro/commit/aa0b577c7ab0241f95ed961715896414f476effa))
* **traduction:** consigne rtl ([5a80d19](https://github.com/SocialGouv/nos1000jours-web-pro/commit/5a80d19d9e7bfbfe66741b0f17554a2a1f1baa1d))
* **traduction:** resultats rtl ([89ce1c5](https://github.com/SocialGouv/nos1000jours-web-pro/commit/89ce1c52e951c04eb57dfd5610f73ea06459c8be))
* **traduction:** rtl sur les résultats ([02358cc](https://github.com/SocialGouv/nos1000jours-web-pro/commit/02358ccd1c6e7e1e100d8f4a6fff3282e338613d))
* **traduction:** taille de la police rtl ([3dd4a50](https://github.com/SocialGouv/nos1000jours-web-pro/commit/3dd4a50501a1766b0070b005792ddd40c98b8f1c))
* **traduction:** taille texte rtl ([75b8874](https://github.com/SocialGouv/nos1000jours-web-pro/commit/75b8874144bd17946b3fa65f51ee1ab34f741428))

## [1.10.5](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.10.4...v1.10.5) (2022-02-02)


### Bug Fixes

* **deps:** update dependency react-bootstrap to v2 ([1956d06](https://github.com/SocialGouv/nos1000jours-web-pro/commit/1956d06f0963675dee4af27eee29c7caf0bf6afa))

## [1.10.4](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.10.3...v1.10.4) (2022-02-02)


### Bug Fixes

* **deps:** update dependency next-i18next to v10 ([d71de88](https://github.com/SocialGouv/nos1000jours-web-pro/commit/d71de88799b23a6bca2b5007453e287097523594))

## [1.10.3](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.10.2...v1.10.3) (2022-02-02)


### Bug Fixes

* **secu:** use non-privilegied docker user ([1ea8c54](https://github.com/SocialGouv/nos1000jours-web-pro/commit/1ea8c54331302ba0cbd5de737a62e77ea848527d))

## [1.10.2](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.10.1...v1.10.2) (2022-01-12)


### Bug Fixes

* **epds:** sauvegarde du test en FR ([2aeb41b](https://github.com/SocialGouv/nos1000jours-web-pro/commit/2aeb41bed0291a56a813ab2c86948855337194da))

## [1.10.1](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.10.0...v1.10.1) (2022-01-11)


### Bug Fixes

* fix config.name ([#60](https://github.com/SocialGouv/nos1000jours-web-pro/issues/60)) ([a7ed055](https://github.com/SocialGouv/nos1000jours-web-pro/commit/a7ed05528d812f25a9245252c43f9846cb02a3cf))

# [1.10.0](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.9.0...v1.10.0) (2022-01-11)


### Features

* **k8s:** add redirection ([2f6610f](https://github.com/SocialGouv/nos1000jours-web-pro/commit/2f6610f9a9e4edbcd971f1ae50ba2ee395def139))

# [1.9.0](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.8.0...v1.9.0) (2022-01-10)


### Features

* **blues:** maj du logo ([6ba4fd0](https://github.com/SocialGouv/nos1000jours-web-pro/commit/6ba4fd04e2cc6adea5c24cee457666efdff0aea3))
* **blues:** renommer le titre ([2a7d558](https://github.com/SocialGouv/nos1000jours-web-pro/commit/2a7d558a2538564dfaca292ca4ba47ae5b3334d2))
* **blues:** suppression menu espace pro ([ee97a1c](https://github.com/SocialGouv/nos1000jours-web-pro/commit/ee97a1c1e989da730801bc49ccba6dfc03075117))

# [1.8.0](https://github.com/SocialGouv/nos1000jours-web-pro/compare/v1.7.1...v1.8.0) (2022-01-05)


### Bug Fixes

* **epds:** retour de PR ([d24b247](https://github.com/SocialGouv/nos1000jours-web-pro/commit/d24b2474dd9dd9664c98a0b21a5d4cdd279248af))
* **epds:** retours PR ([d7fcc6b](https://github.com/SocialGouv/nos1000jours-web-pro/commit/d7fcc6b0caf1f57f51452a760aa8d2890a9d35d5))


### Features


pages/resultats.js
import { useMutation } from "@apollo/client"
import {} from "@dataesr/react-dsfr"
import { useTranslation } from "next-i18next"
import { serverSideTranslations } from "next-i18next/serverSideTranslations"
import { React, useEffect, useState } from "react"
import { Col, Row } from "react-bootstrap"

import { client, EPDS_PARTAGE_INFORMATION } from "../apollo-client"
import { HeaderImage } from "../src/components/HeaderImage"
import { ContentLayout } from "../src/components/Layout"
import { ResultsTab } from "../src/components/ResultsTab"
import {
  LOCAL_IDENTIFIANT_FRANCAIS,
  PATTERN_EMAIL,
  STORAGE_NOM_PATIENT,
  STORAGE_PRENOM_PATIENT,
  STORAGE_RESULTS_BOARD,
  STORAGE_RESULTS_BOARD_TRANSLATED,
  STORAGE_RESULTS_ID,
  STORAGE_RESULTS_LOCALE,
  STORAGE_TOTAL_SCORE,
  URL_1000J,
} from "../src/constants/constants"
import { getInLocalStorage, jsonParse } from "../src/constants/utils"

export default function Resultats() {
  const { t } = useTranslation("resultats")

  return (
    <ContentLayout title={t("header")}>
      <HeaderImage image="/img/bg-espace-pro.png" title={t("header")} />

      <Col className="page-content" style={{ alignItems: "center" }}>
        <h3 className="page-title">{t("resultat")}</h3>
        <FormContact translation={t} />
        <AdsForApp translation={t} />
      </Col>

      <ComprendreTestStyle />
    </ContentLayout>
  )
}

const FormContact = (props) => {
  const [canSend, setCanSend] = useState(false)
  const [isEmailValid, setEmailValid] = useState(false)
  const [isPhoneValid, setPhoneValid] = useState(false)
  const [isEmailProValid, setEmailProValid] = useState(false)
  const [isEmailProSecondaireValid, setEmailProSecondaireValid] =
    useState(false)
  const [queryShareResponses, setQueryShareResponses] = useState()

  const score = getInLocalStorage(STORAGE_TOTAL_SCORE)
  const resultsId = getInLocalStorage(STORAGE_RESULTS_ID)
  const resultsBoard = jsonParse(getInLocalStorage(STORAGE_RESULTS_BOARD))
  const resultsBoardTranslated = jsonParse(
    getInLocalStorage(STORAGE_RESULTS_BOARD_TRANSLATED)
  )
  const resultsLocale = jsonParse(getInLocalStorage(STORAGE_RESULTS_LOCALE))

  const [sendEmailReponseQuery] = useMutation(EPDS_PARTAGE_INFORMATION, {
    client: client,
    onCompleted: () => {
      setQueryShareResponses(props.translation("query-success"))
    },
    onError: (err) => {
      console.warn(err)
      setQueryShareResponses(err.toString())
    },
  })

  const shareEpdsResults = async (inputs) => {
    if (canSend) {
      const name = localStorage.getItem(STORAGE_NOM_PATIENT)
      const surname = localStorage.getItem(STORAGE_PRENOM_PATIENT)

      await sendEmailReponseQuery({
        variables: {
          detail_questions: resultsBoard.map((data) => data.question),
          detail_reponses: resultsBoard.map((data) => data.response),
          detail_score: resultsBoard.map((data) => data.points).map(String),
          email: inputs.inputEmail.value,
          email_pro: inputs.inputEmailPro.value,
          email_pro_secondaire: inputs.inputEmailProSecondaire.value,
          nom: name,
          prenom: surname,
          score: score,
          telephone: inputs.inputTel.value,
          id_reponses: resultsId,
        },
      })
    }
  }

  const send = async (event) => {
    event.preventDefault()
    shareEpdsResults(event.target)
  }

  useEffect(() => {
    setCanSend(isEmailProValid)
  }, [isEmailProValid])

  function handleChange(e) {
    switch (e.target.id) {
      case "inputEmail":
        setEmailValid(e.target.validity.valid)
        break
      case "inputTel":
        setPhoneValid(e.target.validity.valid)
        break
      case "inputEmailPro":
        setEmailProValid(e.target.validity.valid)
        break
      case "inputEmailProSecondaire":
        setEmailProSecondaireValid(e.target.validity.valid)
        break
    }
  }

  return (
    <div>
      <div className="font-weight-bold resultats-score">
        {props.translation("tab.score")} {score} / 30
      </div>
      <p>
        {resultsLocale &&
        resultsLocale.identifiant != LOCAL_IDENTIFIANT_FRANCAIS
          ? `${props.translation("test-completed-in-language")} : ${
              resultsLocale.libelle_francais
            }`
          : null}
      </p>
      <ResultsTab
        translation={props.translation}
        resultsBoard={resultsBoard}
        resultsId={resultsId}
        locale={resultsLocale}
        resultsBoardTranslated={resultsBoardTranslated}
      />

      <div className="font-weight-bold" style={{ marginBottom: 20 }}>
        {props.translation("form.email-pro1-intro")}
      </div>
      <form onSubmit={send}>
        <div
          className={`form-group fr-input-group resultats-form-input ${
            isEmailProValid ? "fr-input-group--valid" : ""
          }`}
        >
          <label>{props.translation("form.email-pro1")}</label>
          <input
            type="email"
            className={`form-control fr-input custom-input ${
              isEmailProValid ? "custom-input-valid" : ""
            }`}
            id="inputEmailPro"
            name="inputEmailPro"
            pattern={PATTERN_EMAIL}
            onChange={handleChange}
            placeholder={props.translation("form.email-pro1-hint")}
            required
          />
        </div>
        <div
          className={`form-group fr-input-group resultats-form-input ${
            isEmailProSecondaireValid ? "fr-input-group--valid" : ""
          }`}
        >
          <label>{props.translation("form.email-pro2")}</label>
          <input
            type="email"
            className={`form-control fr-input custom-input ${
              isEmailProSecondaireValid ? "custom-input-valid" : ""
            }`}
            id="inputEmailProSecondaire"
            name="inputEmailProSecondaire"
            pattern={PATTERN_EMAIL}
            onChange={handleChange}
            placeholder={props.translation("form.email-pro2-hint")}
          />
        </div>
        <div
          className={`form-group fr-input-group resultats-form-input ${
            isEmailValid ? "fr-input-group--valid" : ""
          }`}
        >
          <label>{props.translation("form.email")}</label>
          <input
            type="email"
            className={`form-control fr-input custom-input ${
              isEmailValid ? "custom-input-valid" : ""
            }`}
            id="inputEmail"
            name="inputEmail"
            pattern={PATTERN_EMAIL}
            onChange={handleChange}
            placeholder={props.translation("form.email-hint")}
          />
        </div>
        <div
          className={`form-group fr-input-group resultats-form-input ${
            isPhoneValid ? "fr-input-group--valid" : ""
          }`}
        >
          <label>{props.translation("form.telephone")}</label>
          <input
            type="tel"
            className={`form-control fr-input custom-input ${
              isPhoneValid ? "custom-input-valid" : ""
            }`}
            id="inputTel"
            name="inputTel"
            pattern="[0-9]{10}"
            onChange={handleChange}
            placeholder={props.translation("form.telephone-hint")}
          />
        </div>

        <Col style={{ marginBottom: 20 }}>
          <Row style={{ justifyContent: "center" }}>
            <button
              type="submit"
              className="fr-btn"
              disabled={!canSend}
              style={{ marginTop: 23 }}
            >
              {props.translation("form.envoyer")}
            </button>
          </Row>
          <Row style={{ justifyContent: "center" }}>{queryShareResponses}</Row>
        </Col>
      </form>
    </div>
  )
}

const AdsForApp = ({ translation }) => (
  <div className="resultats-appl-bloc">
    <span className="font-weight-bold" style={{ fontSize: 24 }}>
      {translation("appli.app-1000j")}
    </span>
    <p style={{ lineHeight: "32px", paddingTop: 10 }}>
      {translation("appli.app-1000j-content")}
    </p>
    <div>
      {translation("appli.telecharger")}
      <a
        href={URL_1000J}
        target="_blank"
        style={{ textDecoration: "underline" }}
        rel="noreferrer"
      >
        {URL_1000J}
      </a>
    </div>
    <table style={{ marginTop: 15 }}>
      <tbody>
        <tr>
          <td>
            <img
              alt=""
              src="/img/icone-playstore.svg"
              height={35}
              style={{ marginRight: 20 }}
            />
          </td>
          <td>
            <img alt="" src="/img/icone-appstore.svg" height={35} />
          </td>
        </tr>
        <tr>
          <td>
            <img
              alt=""
              src="/img/QR_google.png"
              height={96}
              style={{ marginRight: 20, marginTop: 10 }}
            />
          </td>
          <td>
            <img
              alt=""
              src="/img/QR_apple.png"
              height={96}
              style={{ marginTop: 10 }}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
)

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale, [
      "common",
      "footer",
      "resultats",
    ])),
  },
})

const ComprendreTestStyle = () => (
  <style jsx="true">{`
    .resultats-form-input {
      max-width: 400px;
    }

    .resultats-score {
      margin-bottom: 40px;
      text-align: center;
      font-size: 24px;
    }

    .resultats-appl-bloc {
      margin-top: 60px;
      padding: 20px 30px 20px 30px;
      background-color: var(--gris);
      border-left: 4px solid var(--bleu-france);
    }
  `}</style>
)
